src/main/drivers/pwm_output.c

```diff
@@ -328,12 +316,27 @@ void motorDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8
         IOConfigGPIO(motors[motorIndex].io, IOCFG_AF_PP);
 #endif
 
+        /* standard PWM outputs */
         if (useUnsyncedPwm) {
-            const uint32_t hz = timerMhzCounter * 1000000;
-            pwmOutConfig(&motors[motorIndex], timerHardware, timerMhzCounter, hz / motorConfig->motorPwmRate, idlePulse, motorConfig->motorPwmInversion);
-        } else {
-            pwmOutConfig(&motors[motorIndex], timerHardware, timerMhzCounter, 0xFFFF, 0, motorConfig->motorPwmInversion);
+            pwmRate = motorConfig->motorPwmRate;
         }
+        const uint32_t clock = timerClock(timerHardware->tim);
+        /* used to find the desired timer frequency for max resolution */
+        const unsigned prescaler = ceilf((clock / pwmRate) / 0xFFFFf); /* 0xFFFF is max period in 16bit timer */
+        const uint32_t hz = clock / prescaler;
+        const unsigned period = hz / pwmRate;
+        
+        const float ticksPerUs = hz / 1e6f; /* aka mhz */
+
+        /* 
+            if brushed then it is the entire length of the period. 
+            TODO: this can be moved back to periodMin and periodLen 
+            once mixer outputs a 0..1 float value.
+        */
+        motors[motorIndex].pulseScale = ((motorConfig->motorPwmProtocol == PWM_TYPE_BRUSHED) ? period : (usLen * ticksPerUs)) / 1000.0f;
+        motors[motorIndex].pulseOffset = (motors[motorIndex].pulseScale * 1000) - (usMin * ticksPerUs);
```

Comment:
Eventually we can get rid of the entire scaling after this is merged and just use 0-1 scaling output from the mixer